### PR TITLE
azurerm_kubernetes_cluster  - add support for the `azure_policy` property in the `addon_profile` block

### PR DESCRIPTION
**Pull request by grahamhayes**
Signed-off-by: Graham Hayes <gr@ham.ie>

### CODE REVIEW

File: azurerm/data_source_kubernetes_cluster.go [hunk 1]
```diff
@@ -77,6 +77,20 @@ func dataSourceArmKubernetesCluster() *schema.Resource {
 								},
 							},
 						},
+
+						"azure_policy": {
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Optional: true,
```
Comment (mbfrahry) [inline azurerm/data_source_kubernetes_cluster.go]: Attributes in DataSources need to be `Computed: true`.  `MaxItems: 1` Can also be removed as it's only used when Creating/Updating a resource.

File: azurerm/data_source_kubernetes_cluster.go [hunk 2]
```diff
@@ -77,6 +77,20 @@ func dataSourceArmKubernetesCluster() *schema.Resource {
...
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enabled": {
+										Type:     schema.TypeBool,
+										Required: true,
```
Comment (mbfrahry) [inline azurerm/data_source_kubernetes_cluster.go]: Also, this should be `Computed: true` instead
```suggestion
										Computed: true,
```

File: azurerm/resource_arm_kubernetes_cluster.go [hunk 3]
```diff
@@ -1043,6 +1057,17 @@ func expandKubernetesClusterAddonProfiles(d *schema.ResourceData) map[string]*co
 		}
 	}
 
+	azurePolicy := profile["azure_policy"].([]interface{})
+	if len(azurePolicy) > 0 {
+		value := azurePolicy[0].(map[string]interface{})
```
Comment (mbfrahry) [inline azurerm/resource_arm_kubernetes_cluster.go]: `azurePolicy[0]` could be nil so we should do a nil check before casting it to `map[string]interface{}`

File: azurerm/resource_arm_kubernetes_cluster.go [hunk 4]
```diff
@@ -1043,6 +1057,17 @@ func expandKubernetesClusterAddonProfiles(d *schema.ResourceData) map[string]*co
...
+		enabled := value["enabled"].(bool)
+
+		addonProfiles["azurepolicy"] = &containerservice.ManagedClusterAddonProfile{
+			Enabled: utils.Bool(enabled),
+			Config:  nil,
```
Comment (mbfrahry) [inline azurerm/resource_arm_kubernetes_cluster.go]: Is Config something we'll want to implement in the future? If so, can you add something like
```suggestion
                         // TODO add support for AddonProfile Config
			Config:  nil,
```

File: website/docs/r/kubernetes_cluster.html.markdown [hunk 5]
```diff
@@ -144,6 +144,7 @@ A `addon_profile` block supports the following:
 * `http_application_routing` - (Optional) A `http_application_routing` block.
 * `oms_agent` - (Optional) A `oms_agent` block. For more details, please visit [How to onboard Azure Monitor for containers](https://docs.microsoft.com/en-us/azure/monitoring/monitoring-container-insights-onboard).
 * `kube_dashboard` - (Optional) A `kube_dashboard` block.
+* `azure_policy` - (Optional) A `azure_policy` block.  For more details please visit [Understand Azure Policy for Azure Kubernetes Service](https://docs.microsoft.com/en-ie/azure/governance/policy/concepts/rego-for-aks)
```
Comment (mbfrahry) [inline website/docs/r/kubernetes_cluster.html.markdown]: We should add a note as well that this attribute is in preview with information for how to opt-in
```suggestion
* `azure_policy` - (Optional) A `azure_policy` block.  For more details please visit [Understand Azure Policy for Azure Kubernetes Service](https://docs.microsoft.com/en-ie/azure/governance/policy/concepts/rego-for-aks)

-> **Note**: Azure Policy for Azure Kubernetes Service is currently in preview and not available to subscriptions that have not [opted-in](https://docs.microsoft.com/en-us/azure/governance/policy/concepts/rego-for-aks?toc=/azure/aks/toc.json) to join `Azure Policy` preview. 
```

File: azurerm/resource_arm_kubernetes_cluster.go (hunk 3 above)
Comment (grahamhayes) [reply azurerm/resource_arm_kubernetes_cluster.go]: sure - I kept it in line with the other policy addons, but I will add a nil check now.

File: azurerm/resource_arm_kubernetes_cluster.go (hunk 4 above)
Comment (grahamhayes) [reply azurerm/resource_arm_kubernetes_cluster.go]: I don't think we will - I think this is always just going to be a true / false flag.

File: azurerm/data_source_kubernetes_cluster.go (hunk 2 above)
Comment (grahamhayes) [reply azurerm/data_source_kubernetes_cluster.go]: Done

File: azurerm/data_source_kubernetes_cluster.go (hunk 1 above)
Comment (grahamhayes) [reply azurerm/data_source_kubernetes_cluster.go]: Done

**Pull request merged**